langkit/textstat.py

```diff
@@ -53,17 +52,21 @@ def wrappee(text: Union[pd.DataFrame, Dict[str, List]]) -> Union[pd.Series, List
 
 
 def init():
-    diagnostic_logger.info("Initialized textstat metrics.")
+    pass
 
-    def unpack(t: Union[Tuple[str, str], Tuple[str, str, str]]) -> Tuple[str, str, str]:
-        return t if len(t) == 3 else (t[0], t[1], t[0])  # type: ignore
 
-    for t in _udfs_to_register:
-        stat_name, schema_name, udf = unpack(t)
-        for column in [lang_config.prompt_column, lang_config.response_column]:
-            register_dataset_udf(
-                [column], udf_name=f"{column}.{udf}", schema_name=schema_name
-            )(wrapper(stat_name, column))
+init()
 
 
-init()
+def _unpack(t: Union[Tuple[str, str], Tuple[str, str, str]]) -> Tuple[str, str, str]:
+    return t if len(t) == 3 else (t[0], t[1], t[0])  # type: ignore
+
+
+for t in _udfs_to_register:
+    stat_name, schema_name, udf = _unpack(t)
+    for column in [lang_config.prompt_column, lang_config.response_column]:
+        register_dataset_udf(
+            [column], udf_name=f"{column}.{udf}", schema_name=schema_name
+        )(wrapper(stat_name, column))
+
+diagnostic_logger.info("Initialized textstat metrics.")
```

Comment:
I think we want only constants/defaults defined as top level in the modules, and most logic for initialization should be encapsulated in the init function, why move the registration outside of init?

Reply:
I wanted to make the registration consistent with the other modules that register at import time via decorators. The goal was to allow `init()` to be called multiple times, so the registration has to happen outside of `init()`.